src/workflows-app/SubmissionConfig.test.js

```diff
@@ -268,25 +280,1411 @@ describe('BaseSubmissionConfig gets proxy urls from Leo', () => {
+    expect(mockSearchResponse).toHaveBeenCalledTimes(1);
+    expect(mockMethodsResponse).toHaveBeenCalledTimes(1);
+
+    // ** ACT **
+    // user selects 'FOO1' record from Data Table
+    const checkboxes = screen.getAllByRole('checkbox');
+    const checkbox = checkboxes[1];
+    await user.click(checkbox);
+
+    // ** ASSERT **
+    // verify that the record was indeed selected
+    expect(checkbox).toHaveAttribute('aria-checked', 'true');
+
+    // ** ACT **
+    const inputsTabButton = await screen.findByRole('button', { name: 'Inputs' });
+    await user.click(inputsTabButton);
+
+    // ** ASSERT **
+    const inputTable = await screen.findByRole('table');
+    const rows = within(inputTable).queryAllByRole('row');
+    expect(rows.length).toBe(runSetInputDef.length + 1); // one row for each input definition variable, plus headers
+
+    // ** ACT **
+    // user sets the source to 'None' for input 'optional_var'
+    const thirdInputRow = within(rows[3]).queryAllByRole('cell');
+    await user.click(within(thirdInputRow[3]).getByText('Type a Value'));
+    const selectOption = await screen.findByText('None');
+    await user.click(selectOption);
+
+    // ** ASSERT **
+    // check that the Attribute column has expected behavior
+    within(thirdInputRow[4]).getByText('Optional');
+
+    // ** ACT **
+    // user clicks on Submit (inputs and outputs should be rendered based on previous submission)
+    const button = screen.getByLabelText('Submit button');
+    await user.click(button);
+
+    // ** ASSERT **
+    // Launch modal should be displayed
+    await screen.getByText('Send submission');
+    const modalSubmitButton = await screen.getByLabelText('Launch Submission');
+
+    // ** ACT **
+    // user click on Submit button
+    await user.click(modalSubmitButton);
+
+    // ** ASSERT **
+    // assert POST /run_sets endpoint was called with expected parameters and input 'optional_var' has correct definition for source 'None'
+    expect(postRunSetFunction).toHaveBeenCalled();
+    expect(postRunSetFunction).toBeCalledWith(
+      cbasUrlRoot,
+      expect.objectContaining({
+        method_version_id: runSetResponse.run_sets[0].method_version_id,
+        workflow_input_definitions: [
+          runSetInputDef[0],
+          runSetInputDef[1],
+          {
+            input_name: 'target_workflow_1.optional_var',
+            input_type: {
+              optional_type: {
+                primitive_type: 'String',
+                type: 'primitive',
+              },
+              type: 'optional',
+            },
+            source: {
+              type: 'none',
+            },
+          },
+        ],
+        workflow_output_definitions: runSetOutputDef,
+        wds_records: {
+          record_type: 'FOO',
+          record_ids: ['FOO1'],
+        },
+      })
+    );
+  });
+
+  it('should call POST /run_sets endpoint with expected parameters after struct has been updated', async () => {
```

Comment:
@nawatts This test is occasionally timing out on CircleCI - are there any glaring red flags here or is it simply too long? I think a lot of the time is being sucked up by `user.clear` and `user.type`, which I put in place to reduce our usage of `fireEvent`

Reply:
It does look pretty long.

It also does lots of unnecessary `await`ing. `getBy...` and `queryBy...` queries do not return a Promise, so they do not need to be awaited. Also curious if some of the `findBy...` queries (which are async) could be replaced by `getBy...`.

If nothing else, the default timeout of 5s can be overridden by passing a third argument to `it`.
https://jestjs.io/docs/27.x/api#testname-fn-timeout